protobuf/lib/src/protobuf/pb_list.dart

```diff
@@ -200,4 +204,26 @@ class PbList<E> extends ListBase<E> {
   static Never _readOnlyError(String methodName) {
     throw UnsupportedError("'$methodName' on a read-only list");
   }
+
+  PbList<E> deepCopy({bool freeze = false}) {
+    // TODO: We know the capacity, is it possible to allocate a list with the right size?
+    final wrappedList = <E>[];
```

Comment:
You could possibly use `List.generate` to fill in the right values from the start, thereby avoiding the re-allocation.